server/worker/jira.go

```diff
@@ -42,73 +50,256 @@ View the affected software and more affected hosts:
 ----
 
 This issue was created automatically by your Fleet Jira integration.
-`,
-))
+`)),
 
-type jiraTemplateArgs struct {
+	FailingPolicySummary: template.Must(template.New("").Parse(
+		`{{ .PolicyName }} policy failed on {{ len .Hosts }} host(s)`,
+	)),
+
+	FailingPolicyDescription: template.Must(template.New("").Parse(
+		`Hosts:
+{{ $end := len .Hosts }}{{ if gt $end 50 }}{{ $end = 50 }}{{ end }}
+{{ range slice .Hosts 0 $end }}
+* [{{ .Hostname }}|{{ $.FleetURL }}/hosts/{{ .ID }}]
+{{ end }}
+
+View hosts that failed {{ .PolicyName }} on the [*Hosts*|{{ .FleetURL }}/hosts/manage/?order_key=hostname&order_direction=asc&{{ if .TeamID }}team_id={{ .TeamID }}&{{ end }}policy_id={{ .PolicyID }}&policy_response=failing] page in Fleet.
+
+----
+
+This issue was created automatically by your Fleet Jira integration.
+`)),
+}
+
+type jiraVulnTplArgs struct {
 	NVDURL   string
 	FleetURL string
 	CVE      string
 	Hosts    []*fleet.HostShort
 }
 
+type jiraFailingPoliciesTplArgs struct {
+	FleetURL   string
+	PolicyID   uint
+	PolicyName string
+	TeamID     *uint
+	Hosts      []fleet.PolicySetHost
+}
+
 // JiraClient defines the method required for the client that makes API calls
 // to Jira.
 type JiraClient interface {
-	CreateIssue(ctx context.Context, issue *jira.Issue) (*jira.Issue, error)
+	CreateJiraIssue(ctx context.Context, issue *jira.Issue) (*jira.Issue, error)
+	JiraConfigMatches(opts *externalsvc.JiraOptions) bool
 }
 
 // Jira is the job processor for jira integrations.
 type Jira struct {
-	FleetURL   string
-	Datastore  fleet.Datastore
-	Log        kitlog.Logger
-	JiraClient JiraClient
+	FleetURL      string
+	Datastore     fleet.Datastore
+	Log           kitlog.Logger
+	NewClientFunc func(*externalsvc.JiraOptions) (JiraClient, error)
+
+	// mu protects concurrent access to clientsCache, so that the job processor
+	// can potentially be run concurrently.
+	mu sync.Mutex
```

Comment:
The `worker` does not run jobs concurrently at the moment, but I didn't want the _job processors_ to have that assumption baked in (so that if someday we change the worker, the job processors will still work fine). And it's not that complex to support.